.github/workflows/e2e.yml

```diff
@@ -0,0 +1,63 @@
+name: Karina upstream
+
+on:
+  pull_request:
+    branches: [ master, main ]

```

Comment:
Why the rename and addition of main ?

Reply:
I add main by reflex to CIs for forward compat on a rename, if we ever choose to do one.

The job rename is because the current job is a karina upstream test and adding the go tests to that didn't feel like I was putting them in the right spot.